libnavigation-util/src/main/java/com/mapbox/navigation/utils/extensions/HandlerThreadEx.kt

```diff
@@ -0,0 +1,14 @@
+@file:JvmName("HandlerThreadEx")
+
+package com.mapbox.navigation.utils.extensions
+
+import android.os.Build
+import android.os.HandlerThread
+
+fun HandlerThread.quitSafelySupport() {
```

Comment:
Same as above re: `internal` and `internal` package.

In general, shouldn't all the internal extensions be `internal` and under the `internal package`?

Reply:
I don't think so, because we have separate `utils` module for `utils` and `extensions`. Moreover if inside module (for example `metric` module) we need only one specific extension linked only with this module (for example `MetricEx`), I think it can be placed inside this module. WDYT?

Reply:
Yeah, if we're going to use them across different modules is 🆗 if we don't add the `internal` modifier but we should add them under the `internal` package so developers know that they don't need to rely on them and also that can be changed "without" breaking SemVer.

Reply:
it's a part of util` module, that we don't expose and nobody could use one, except us

Reply:
But as I know, `utils` module is only for our usage, because we add it to other modules via `implementation` keyword in gradle, not `api`. So we can mark all utils as public to use it everywhere we want, isn't it?

Reply:
What do you mean that we don't expose? At the moment that we include it as part of the dependencies 👀 https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/build.gradle#L60 the APIs are available.

Reply:
It's `implementation` already, and it's leaking. Only way we have here is to create an `internal` package and place all these utils in there as I mentioned above. This is what we currently do in `libandroid-navigation` and we should follow the same approach in the rest of the modules.

Reply:
can't agree, when we build dependencies as `implementation` than only partucular module has access to it

Reply:
As I said, problem is that the final artifact (`aar`) has access to it, independently of using `implementation` or `api`. Maybe I'm missing something, though. Could you run a quick test generating an artifact that has a dependency as `implementation` and then another one having it as `api` and compare / see if for both classes are accessible?